### PR TITLE
add perk Armorcraft from Engineering tree

### PR DESCRIPTION
**Pull request by miguelcjalmeida**
I added the Armorcraft perk that increases character's armor by 10%. I had just to postfix the Agent.GetBaseArmorEffectivenessForBodyPart, to increase its output by 10% when the Agent had the perk active.

### CODE REVIEW

**Comment by Tyler-IN** [comment]
Don't forget to update README.md

File: src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ArmorcraftPatch.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    // ReSharper disable once InconsistentNaming
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void Postfix(ref float __result, ref Agent __instance) {
+      if (!__instance.IsHuman) return;
+      if (!__instance.IsHero) return;
+      if (!(__instance.Character is CharacterObject)) return;
+
+      var character = (CharacterObject)__instance.Character;
+      var hero = character?.HeroObject;
+      if (hero == null) return;
```
Comment (fnzr) [inline src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ArmorcraftPatch.cs:82]: There's no need for all these checks
```
if (__instance.IsHero && __instance.Character is CharacterObject charObj) {
 var hero = charObj.HeroObject;
 //apply perk
}
```
is enough.

Comment (Tyler-IN) [reply src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ArmorcraftPatch.cs:82]: I think at least the `IsHuman` check should probably be removed here. I don't know if any assertion that the character is human is valid for the armorcraft perk?

Comment (miguelcjalmeida) [reply src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ArmorcraftPatch.cs:82]: Without checking IsHuman first, the code crashes on the next line when checking if it is a hero. You could test this by simply jumping with your horse.

Comment (fnzr) [reply src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ArmorcraftPatch.cs]: Really? That's really unexpected. I wonder if there's an underlying issue on the TW code, a boolean test like that shouldn't crash.

**Pull request merged**